### PR TITLE
rework @loadable integration (dev mode server reloads, HMR)

### PR DESCRIPTION
**Pull request by tadeuszwojcik**
rework @loadable integration with server rebuilds on change and hot reloading 
- requires pull request https://github.com/smooth-code/loadable-components/pull/187 to be merged and @loadable components dependencies version update 
- currently server.test.js tests are failing, that will probably require mocking ./build/loadable-stats.json file somehow

### CODE REVIEW

File: packages/falcon-client/build-utils/paths.js [hunk 1]
```diff
@@ -53,10 +53,7 @@ module.exports = {
   appBuildPublic: resolveApp('build/public'),
   appPublic: resolveApp('public'),
   appManifest: resolveApp('build/assets.json'),
-  // TODO: this path should be changed to /build/loadable-stats.json
-  // when https://github.com/smooth-code/loadable-components/issues/179 gets fixed
-  // as we don't want to have stats file in public dir in long term
-  loadableStats: resolveApp('build/public/loadable-stats.json'),
+  loadableStats: resolveApp('build/loadable-stats.json'),
```
Comment (tutok) [inline packages/falcon-client/build-utils/paths.js]: we should follow convention I think and `loadableStats` should be renamed to `appLoadableStats`

Comment (tutok) [reply packages/falcon-client/build-utils/paths.js]: done

File: packages/falcon-client/src/server.test.js [hunk 2]
```diff
@@ -111,6 +111,7 @@ describe('Server', () => {
     }).callback();
     const response = await supertest(serverHandler).get('/');
 
+    console.log(response);
```
Comment (Borales) [inline packages/falcon-client/src/server.test.js]:
```suggestion
```

File: yarn.lock [hunk 3]
```diff
@@ -2228,7 +2228,15 @@
...
-"@types/react@*", "@types/react@16.7.17":
+"@types/react@*":
+  version "16.7.20"
+  resolved "https://registry.npmjs.org/@types/react/-/react-16.7.20.tgz#13ae752c012710d0fa800985ca809814b51d3b58"
+  integrity sha512-Qd5RWkwl6SL7R2XzLk/cicjVQm1Mhc6HqXY5Ei4pWd1Vi8Fkbd5O0sA398x8fRSTPAuHdDYD9nrWmJMYTJI0vQ==
+  dependencies:
+    "@types/prop-types" "*"
+    csstype "^2.2.0"
+
+"@types/react@16.7.17":
```
Comment (Borales) [inline yarn.lock]: @tutok please check which package still uses 16.7.17

File: packages/falcon-client/src/server.test.js (hunk 2 above)
Comment (tutok) [reply packages/falcon-client/src/server.test.js]: yes see that, I am working with tests actually

**Pull request merged**